src/api/localStorage.js

```diff
@@ -0,0 +1,68 @@
+export class LocalStorage {
+  constructor() {
+    this.users = JSON.parse(
+      localStorage.getItem('UserAccounts')) || [];
+  }
+
+  createUser(user) {
+    user.id = this.generateId();
+    this.users.push(user);
+    this.syncStorage();
+  }
+
+  syncStorage() {
+    localStorage.setItem('UserAccounts', JSON.stringify(this.users));
+    this.updateUsers();
+  }
+
+  updateUsers() {
+    this.users = JSON.parse(localStorage.getItem('UserAccounts')) || [];
+  }
+
+  getUser(id) {
+    const user = this.users.find(user => String(user.id) === String(id));
+    return user;
+  }
+
+  getAllUsers() {
+    return this.users;
+  }
+
+  updateUser(id, update) {
+    try {
+      const updatedUser = Object.assign(this.getUser(id), update);
+      this.users.splice(
+        this.users.findIndex(
+          user => String(user.id) === id), 1, updatedUser);
+      this.syncStorage();
+    } catch (err) {
+      console.error('Error Editing User: ', err);
+    }
+  }
+
+  deleteUser(id) {
+    const userIndex = this.users.findIndex(user => String(user.id) === id);
+    this.users.splice(userIndex, 1);
+    this.syncStorage();
+  }
+
+  deleteAllUsers() {
+    this.users = [];
+    localStorage.clear();
```

Comment:
tiny nit: you may want to ensure `localStorage.clear()` has run before resetting `this.users`, in the event the clear doesn't actually complete.